mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageMapper.java

```diff
@@ -311,8 +311,10 @@ public MessageMetaData add(Mailbox mailbox, MailboxMessage message) throws Mailb
     public List<UpdatedFlags> resetRecent(Mailbox mailbox) {
         CassandraId mailboxId = (CassandraId) mailbox.getMailboxId();
 
-        Flux<ComposedMessageIdWithMetaData> toBeUpdated = messageIdDAO
-            .retrieveMessages(mailboxId, MessageRange.all(), Limit.unlimited())
+        Flux<ComposedMessageIdWithMetaData> toBeUpdated = mailboxRecentDAO.getRecentMessageUidsInMailbox(mailboxId)
+            .collectList()
+            .flatMapMany(uids -> Flux.fromIterable(MessageRange.toRanges(uids)))
+            .concatMap(range -> messageIdDAO.retrieveMessages(mailboxId, MessageRange.all(), Limit.unlimited()))
```

Comment:
you don't use `range`, so there is the same issue (but easier to fix now ^^)

Reply:
oups